CONTRIBUTING.md

```diff
@@ -110,7 +110,8 @@ cmake -DUNIT_TESTING=ON -DUNIT_TESTING_MOCKS=ON ..
 
 ### Build Docs
 
-Running below command from root folder will create a new folder `docs` containing html file with documentation about CORE headers. Make sure you have `doxygen` installed on the system.
+Running below command from root folder will create a new folder `docs` containing html file with documentation about CORE headers. Make sure you have `doxygen` version *1.8.18* or later installed on the system.
```

Comment:
Should we mention some version info at the root read me, here?
https://github.com/Azure/azure-sdk-for-c/tree/660ed24269266c16f6f09cbd9aa315b5a8e1253f#debian

What version will folks get when they run this, and how can they control it?
```
sudo apt install doxygen
```

Similarly, maybe we should add a link to the doxygen release for folks to go install it on Windows at the root README, just like we have it in this contributing doc:
https://github.com/Azure/azure-sdk-for-c/tree/660ed24269266c16f6f09cbd9aa315b5a8e1253f#windows

https://www.doxygen.nl/download.html

Reply:
it depends on the definition of a development environment right?
it currently doesn't mention anything about adding doxygen, so, changing this would be better address on a PR to improve the dev env documentation.

Also, we shouldn't worry about explaining how to install/update dependencies. I went on that path before but is creates a relation that needs to keep updated. For every change on the dependency, our docs might become obsolete. 
So, we should focus on explaining about the SDK and just mention what dependencies we require.